backstack/src/commonMain/kotlin/com/slack/circuit/backstack/SaveableBackStack.kt

```diff
@@ -38,6 +38,10 @@ public class SaveableBackStack : BackStack<SaveableBackStack.Record> {
 
   private val entryList = mutableStateListOf<Record>()
 
+  // TODO: make this lazy?
+  // TODO: make this map limited in size?
```

Comment:
I think I'll just leave store unbounded. The chances of people filling this up to unimaginable sizes are extremely slim.

Using `SnapshotStateMap`. We need one of those as nothing observes the state store. It is purely a place to stuff items and pull them back again.

> can we ever decide when to clear saved stores

Maybe? I think using `saveState = false` and `restoreState = false` is more defined by the UX, rather than something we need to write logic for. I can't think of a good heuristic for clearing out the state anyway.